spec/producer_spec.rb

```diff
@@ -360,6 +364,51 @@
     end
   end
 
+  describe '#interceptor' do
+    let(:now) { Time.now }
+
+    it "creates and shuts down a producer with interceptor" do
+      interceptor = FakeProducerInterceptor.new
+      producer = initialize_producer(
+        interceptors: [interceptor],
+        cluster: cluster,
+        transaction_manager: transaction_manager
+      )
+
+      producer.shutdown
+      expect(interceptor.close_count).to eq 1
+    end
+
+    it "does not break the on_send chain" do
+      interceptor1 = FakeProducerInterceptor.new(append_s: 'hello2', on_send_error: true)
+      interceptor2 = FakeProducerInterceptor.new(append_s: 'hello3')
+      interceptors = Kafka::ProducerInterceptors.new(interceptors: [interceptor1, interceptor2], logger: logger)
+      intercepted_message = interceptors.on_send(Kafka::PendingMessage.new(
+        value: "hello1",
+        key: nil,
+         headers: {
+           hello: 'World'
+         },
+         topic: "greetings",
+         partition: 0,
+         partition_key: nil,
+         create_time: now
+      ))
+
+      expect(intercepted_message).to eq Kafka::PendingMessage.new(
+        value: "hello1hello3",
+        key: nil,
+        headers: {
+          hello: 'World'
+        },
+        topic: "greetings",
+        partition: 0,
+        partition_key: nil,
+        create_time: now
+      )
+    end
+  end
```

Comment:
do you mean that if that code was in [producer_interceptors.rb](https://github.com/Shopify/ruby-kafka/blob/c06f4bfcdfd23dcbce8317a2fd8371b071c619ca/lib/kafka/producer_interceptors.rb#L16-L27) instead of the current code, the tests would still pass?
they do not

Reply:
added more tests